tests/testthat/test-qenv_eval_code.R

```diff
@@ -88,22 +88,25 @@ testthat::test_that("an error when calling eval_code returns a qenv.error object
     unname(q$trace),
     c("x <- 1", "y <- 2", "z <- w * x")
   )
-  testthat::expect_equal(q$message, "object 'w' not found \n when evaluating qenv code:\nz <- w * x")
+  testthat::expect_equal(q$message, "object 'w' not found \n when evaluating qenv code:\nexpression(z <- w * x)")
 })
 
 testthat::test_that("a warning when calling eval_code returns a qenv object which has warnings", {
   q <- eval_code(qenv(), quote("iris_data <- iris"))
   q <- eval_code(q, quote("p <- hist(iris_data[, 'Sepal.Length'], ff = '')"))
   testthat::expect_s4_class(q, "qenv")
   testthat::expect_equal(
-    q@warnings,
-    c("", paste(rep("> \"ff\" is not a graphical parameter\n", 4), collapse = ""))
+    lapply(q@code, attr, "warning"),
+    list(NULL, "> \"ff\" is not a graphical parameter\n")
   )
 })
 
 testthat::test_that("eval_code with a vector of code produces one warning element per code element", {
```

Comment:
description no longer relevant. Warnings are the attribute of the code elements. It rather should be
`eval_code associates warnings with call by adding attribute to code element`